src/Streamly/External/ByteString.hs

```diff
@@ -21,12 +21,12 @@ import Foreign.ForeignPtr.Compat (plusForeignPtr)
 #endif
 import Foreign.ForeignPtr.Unsafe (unsafeForeignPtrToPtr)
 import GHC.Ptr (minusPtr, plusPtr)
-import Streamly.Internal.Memory.Array.Types (Array(..))
-import Streamly.Internal.Data.Unfold.Types (Unfold(..))
+import Streamly.Internal.Data.Array.Foreign.Type (Array(..))
+import Streamly.Internal.Data.Fold (Fold)
+import Streamly.Internal.Data.Unfold.Type (Unfold)
 import Streamly.Internal.Data.Unfold (lmap)
-import Streamly.Internal.Data.Fold.Types (Fold(..))
 
-import qualified Streamly.Internal.Memory.Array as A
+import qualified Streamly.Internal.Data.Array.Foreign as Array
```

Comment:
Can we import Streamly.Data.Array.Foreign instead? And just import selected APIs from Internal if any.